akka-http-core/src/test/scala/akka/http/impl/engine/client/ConnectionPoolSpec.scala

```diff
@@ -228,6 +228,45 @@ class ConnectionPoolSpec extends AkkaSpec("""
       acceptIncomingConnection()
       val (Success(_), 42) = responseOut.expectNext()
     }
+
+    "never close hot connections when minConnections key is given and >0" in new TestSetup() {
```

Comment:
That would simplify the code a lot, but I wonder what would be the assumption here? `acceptIncomingConnection()` didn't throw? This outline seems straightforward, but I'm not sure if this is enough to validate that the internal state is correct ... What do you think?